transcriptomics_data_service/main.py

```diff
@@ -32,13 +36,14 @@ async def lifespan(_app: FastAPI):
 
 
 app = BentoFastAPI(
-    authz_middleware=None,
+    authz_middleware=authz_plugin,
     config=config_for_setup,
     logger=logger_for_setup,
     bento_extra_service_info=BENTO_SERVICE_INFO,
     service_type=SERVICE_TYPE,
     version=__version__,
     lifespan=lifespan,
+    dependencies=authz_plugin.dep_app(),
```

Comment:
I find it confusing that this dependency is a list of a single dependency instead of one dependency, is there a reason for that?

Reply:
Yeah I went back and forth with this.
The FastAPI `dependencies` arguments (in app, routers and endpoints) can take `None | Sequence[Depends]`.
I decided to make `BaseAuthzMiddleware.dep_<specific dep location>` return the same, so that implementers could provide a sequence of dependencies if they wish.
The only exception being `dep_public_endpoint`, since `BentoFastAPI` expects a single dependency here.

Reply:
I think that allowing implementers to include dependency sequences gives them more flexibility.
For instance, a dependency sequence could perform authz checks by:
1. Decoding a JWT
2. Looking for the presence of a a claim in the token's content to authorize

Reply:
sounds good